src/composition/supergraph/config/full/supergraph.rs

```diff
@@ -108,6 +108,11 @@ impl FullyResolvedSupergraphConfig {
         self.subgraphs.insert(name, subgraph)
     }
 
+    /// Gets a copy of a subgraph by name
+    pub fn get_subgraph(&mut self, name: &str) -> Option<FullyResolvedSubgraph> {
+        self.subgraphs.get(name).cloned()
+    }
```

Comment:
This is to enable us to get a copy of a subgraph from a live `SupergraphConfig`, then update its routing_url and put it back again, as update requires us to have the whole subgraph. I tried a version that used `get_mut` but this caused FileWatchers to be dropped amongst other things and debugging that seemed harder than just using this slightly more round-a-bout method